### PR TITLE
Add a --heron-home option into heron-cli to specify lib/config directory

### PR DESCRIPTION
**Pull request by nwangtw**
Inside Twitter we need to support multi heron version and we need to be able to override paths to heron libs/configs in the internal tools.

This PR just refactor the two get_*_heron_dir() functions slightly so that the paths are available to be overridden beforehand if needed.

### CODE REVIEW

File: heron/tools/common/src/python/utils/config.py [hunk 1]
```diff
@@ -58,6 +58,12 @@
 ROLE_KEY = "role.required"
 ENVIRON_KEY = "env.required"
 
+# global variabls that can be overriden when necessary.
+# Note that overriden has to happen before the first call
```
Comment (huijunw) [inline heron/tools/common/src/python/utils/config.py]: who overrides them

Comment (nwangtw) [reply heron/tools/common/src/python/utils/config.py]: Users who want to customize the tools.

Comment (nwangtw) [reply heron/tools/common/src/python/utils/config.py]: I can also add an option to override heron client data files in cli.

**Comment by nwangtw** [comment]
I am planning to add an option in heron client to specify this path. It could be useful for all users and it doesn't have side effects.

**Comment by nwangtw** [comment]
Probably the last PR for this year~

**Comment by nwangtw** [comment]
Dropping this PR in favor of easier installation of older build. Heron tools package is self contained basically so installation is easy.

**Pull request closed without merging**